druid/src/widget/focus.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 The druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A focus widget.
+
+use druid::widget::prelude::*;
+
+use druid::{
+    commands, Command, Data, FocusNode, HotKey, KbKey, Point, Rect, SysMods, Widget, WidgetPod,
+};
+
+/// A widget that manages a FocusNode to allow focus to be given to this widget and its descendants.
+pub struct Focus<T> {
+    child: WidgetPod<T, Box<dyn Widget<T>>>,
+    auto_focus: bool,
+    focus_requested: bool,
+    focus_node: FocusNode,
+}
+
+impl<T: Data> Focus<T> {
+    /// Create a new Focus widget with a child
+    pub fn new(child: impl Widget<T> + 'static) -> Self {
+        Focus {
+            child: WidgetPod::new(child).boxed(),
+            focus_node: FocusNode::empty(),
+            auto_focus: false,
+            focus_requested: false,
+        }
+    }
+
+    /// Builder-style method to set the `Focus`'s auto focus.
```

Comment:
What does 'auto focus' mean? (I presume it means "has focus when the program starts", but we should be explicit about this.) What happens when multiple widgets are auto focused?

Reply:
> has focus when the program starts

this. I'll try to make more understandable the doc comment.